includes/admin/payments/class-refund-items-table.php

```diff
@@ -125,30 +117,140 @@ public function get_sortable_columns() { return array(); }
+			$name = $item->get_order_item_name();
+		} elseif ( $item instanceof Order_Adjustment ) {
+			$default = $item->type === 'credit' ? __( 'Order Credit', 'easy-digital-downloads' ) : __( 'Order Fee', 'easy-digital-downloads' );
+			$name    = ! empty( $item->description ) ? $item->description : $default;
+		}
+
+		return $name;
 	}
 
 	/**
 	 * This function renders most of the columns in the list table.
 	 *
 	 * @since 3.0
 	 *
-	 * @param \EDD\Orders\Order_Item $order_item  Order item object.
-	 * @param string                 $column_name The name of the column.
+	 * @param Order_Item|Order_Adjustment $item        Order item or adjustment object.
+	 * @param string                      $column_name The name of the column.
 	 *
 	 * @return string Column name.
 	 */
-	public function column_default( $order_item, $column_name ) {
+	public function column_default( $item, $column_name ) {
+		$object_type   = $this->get_object_type( $item );
+		$item_id       = $this->get_item_unique_id( $item );
+		$item_quantity = $item instanceof Order_item ? $item->quantity : 1;
+
 		switch ( $column_name ) {
 			case 'amount':
-			case 'discount':
-			case 'tax':
-			case 'subtotal':
-			case 'total':
-				return $this->format_currency( $order_item, $column_name );
+				return $this->format_currency( $item, $column_name );
+
+			case 'total' :
+				return $this->format_currency( $item, $column_name, 0 );
+
+			case 'quantity' :
+				ob_start();
+				?>
+				<label for="edd-order-item-quantity-<?php echo esc_attr( $item_id ); ?>" class="screen-reader-text">
+					<?php esc_html_e( 'Quantity to refund', 'easy-digital-downloads' ); ?>
+				</label>
+				<input type="number" id="edd-order-item-quantity-<?php echo esc_attr( $item_id ); ?>" class="edd-order-item-refund-quantity edd-order-item-refund-input" name="refund_<?php echo esc_attr( $object_type ); ?>[<?php echo esc_attr( $item->id ); ?>][quantity]" value="<?php echo esc_attr( $item_quantity ); ?>" placeholder="0" min="0" max="<?php echo esc_attr( $item_quantity ); ?>" step="1" disabled />
+				<?php
+				return ob_get_clean();
+
+			case 'subtotal' :
+			case 'tax' :
+				$currency_pos = edd_get_option( 'currency_position', 'before' );
+
+				// Maximum amounts that can be refunded.
+				$refundable_amounts = $item->get_refundable_amounts();
+				$amount_remaining   = array_key_exists( $column_name, $refundable_amounts ) ? $refundable_amounts[ $column_name ] : $item->{$column_name};
+
+				/*
+				 * Original amount.
+				 * For subtotals, we actually do subtotal minus discounts for simplicity so that the end user
+				 * doesn't have to juggle that.
+				 */
+				$original_amount = $item->{$column_name};
+				if ( 'subtotal' === $column_name && ! empty( $item->discount ) ) {
+					$original_amount -= $item->discount;
+				}
+				ob_start();
+				?>
+				<label for="edd-order-item-<?php echo esc_attr( $item_id ); ?>-refund-<?php echo esc_attr( $column_name ); ?>" class="screen-reader-text">
```

Comment:
Could we update this cell markup to use the new form group markup? It would make it more manageable for styling. Something like this:

```php
				<div class="edd-form-group">
					<label for="edd-form-group__label edd-order-item-<?php echo esc_attr( $item_id ); ?>-refund-<?php echo esc_attr( $column_name ); ?>" class="screen-reader-text">
						<?php
						if ( 'subtotal' === $column_name ) {
							esc_html_e( 'Amount to refund, excluding tax', 'easy-digital-downloads' );
						} else {
							esc_html_e( 'Amount of tax to refund', 'easy-digital-downloads' );
						}
						?>
					</label>
					<div class="edd-form-group__control">
						<?php
						if ( 'before' === $currency_pos ) {
							echo '<span class="edd-amount-control__currency is-before">';
							echo esc_html( $this->get_currency_symbol( $item->order_id ) );
							echo '</span>';
						}
						?>
						<input type="text" id="edd-order-item-<?php echo esc_attr( $item_id ); ?>-refund-<?php echo esc_attr( $column_name ); ?>" class="edd-order-item-refund-<?php echo esc_attr( $column_name ); ?> edd-order-item-refund-input" name="refund_<?php echo esc_attr( $object_type ); ?>[<?php echo esc_attr( $item->id ); ?>][<?php echo esc_attr( $column_name ); ?>]" value="<?php echo esc_attr( edd_format_amount( $amount_remaining ) ); ?>" placeholder="<?php echo esc_attr( edd_format_amount( 0 ) ); ?>" data-original="<?php echo esc_attr( $original_amount ); ?>" data-max="<?php echo esc_attr( $amount_remaining ); ?>" disabled />
						<?php
						if ( 'after' === $currency_pos ) {
							echo '<span class="edd-amount-control__currency is-after">';
							echo esc_html( $this->get_currency_symbol( $item->order_id ) );
							echo '</span>';
						}
						?>
					</div>
					<small class="edd-order-item-refund-max-amount">
						<?php
						echo _x( 'Max:', 'Maximum input amount', 'easy-digital-downloads' ) . '&nbsp;';

						echo $this->format_currency( $item, $column_name, $amount_remaining );
						?>
					</small>
				</div>
```